services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -2871,41 +2882,53 @@ private Pair<Boolean, Version> addVersion(
                   useFastKafkaOperationTimeout);
             }
 
+            // We shouldn't need to create view resources in parent fabric
+            if (!multiClusterConfigs.isParent()) {
+              Properties veniceViewProperties = new Properties();
+              veniceViewProperties.put(PARTITION_COUNT, numberOfPartitions);
+              veniceViewProperties.put(USE_FAST_KAFKA_OPERATION_TIMEOUT, useFastKafkaOperationTimeout);
+              veniceViewProperties.putAll(clusterConfig.getProps().toProperties());
+              veniceViewProperties.put(LOG_COMPACTION_ENABLED, false);
+              veniceViewProperties.put(KAFKA_REPLICATION_FACTOR, clusterConfig.getKafkaReplicationFactor());
+              veniceViewProperties.put(ETERNAL_TOPIC_RETENTION_ENABLED, true);
+
+              constructViewResources(veniceViewProperties, store, version, compressionDictionary);
+            }
+
             if (sendStartOfPush) {
               ByteBuffer compressionDictionaryBuffer = null;
               if (compressionDictionary != null) {
                 compressionDictionaryBuffer =
                     ByteBuffer.wrap(EncodingUtils.base64DecodeFromString(compressionDictionary));
-              } else if (store.getCompressionStrategy().equals(CompressionStrategy.ZSTD_WITH_DICT)) {
+              } else if (version.getCompressionStrategy().equals(CompressionStrategy.ZSTD_WITH_DICT)) {
                 // This compression strategy needs a dictionary even if there is no input data,
                 // so we generate a dictionary based on synthetic data. This is done in vpj driver
                 // as well, but this code will be triggered in cases like Samza batch push job
                 // which is independent of the vpj flow.
                 compressionDictionaryBuffer = emptyPushZSTDDictionary.get();
               }
 
-              final Version finalVersion = version;
               VeniceWriter veniceWriter = null;
               try {
                 VeniceWriterOptions.Builder vwOptionsBuilder =
-                    new VeniceWriterOptions.Builder(finalVersion.kafkaTopicName()).setUseKafkaKeySerializer(true)
+                    new VeniceWriterOptions.Builder(version.kafkaTopicName()).setUseKafkaKeySerializer(true)
                         .setPartitionCount(numberOfPartitions);
-                if (multiClusterConfigs.isParent() && finalVersion.isNativeReplicationEnabled()) {
+                if (multiClusterConfigs.isParent() && version.isNativeReplicationEnabled()) {
                   // Produce directly into one of the child fabric
-                  vwOptionsBuilder.setBrokerAddress(finalVersion.getPushStreamSourceAddress());
+                  vwOptionsBuilder.setBrokerAddress(version.getPushStreamSourceAddress());
                 }
                 veniceWriter = getVeniceWriterFactory().createVeniceWriter(vwOptionsBuilder.build());
```

Comment:
Do we have logic to send `SOP` to view topics? And `sort` flag should be always `false` for view topics.

Reply:
Good catch, we currently are using the same method to write `SOP` for VT. I think it's worth it to create a new one explicitly for view topics and we always set `sort` flag to `false`.